### DISCUSSION TITLE
whats this ,how to deal with it

### DISCUSSION THREAD

**Discussion by xulegithub** [infiniflow/ragflow#2591] [General]
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.41.34120\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for fasttext
Failed to build fasttext
ERROR: Could not build wheels for fasttext, which is required to install pyproject.toml-based projects

**Comment by JinHai-CN** [comment]
I suppose you are trying to build RAGFlow source code, am I right?

**Comment by xulegithub** [comment]
yes you are right i am trying to build RAGFlow on local laptop ,but i cant handle this problem,so i senserly need help .thank you very much!
________________________________
发件人: Jin Hai ***@***.***>
发送时间: 2024年9月24日 10:10
收件人: infiniflow/infinity ***@***.***>
抄送: xule ***@***.***>; Author ***@***.***>
主题: Re: [infiniflow/infinity] whats this ,how to deal with it (Discussion infiniflow/infinity#1909)


I suppose you are trying to build RAGFlow source code, am I right?

―

**Comment by JinHai-CN** [comment]
We usually build RAGFlow on Linux. You can try to install WSL on Windows to build RAGFlow.

**Reply by xulegithub** [reply]
ok thank you !!!!
